re #1025 - API Catalog: Ability to search for and filter with multiple tags

kahboom opened the pull request:
Changes:
- Add tag filter that supports multiple tags. Gather tags from available APIs. Sort tags alphabetically.
- Ability to clear all tag selections or individual tags.
- Support for all filters in combinations.
- Add UI-select tagging ability to search for any tags.
- CSS to match neighboring API type dropdown.

Tested on Chrome, Safari & FF running OS X El Capitan.

--

### Screenshots

**Default view:**
<img width="1174" alt="screen shot 2016-03-18 at 12 46 36 pm" src="https://cloud.githubusercontent.com/assets/3844502/13885901/ed9ff3ae-ed0a-11e5-98be-ab72fdd8aaaa.png">

**Focus opens up dropdown (see note about this below):**
<img width="1185" alt="screen shot 2016-03-18 at 12 46 51 pm" src="https://cloud.githubusercontent.com/assets/3844502/13885904/f16b23a0-ed0a-11e5-8477-e350498fcac0.png">

**Filter with one tag selected:**
<img width="1174" alt="screen shot 2016-03-18 at 12 47 06 pm" src="https://cloud.githubusercontent.com/assets/3844502/13885906/f711334e-ed0a-11e5-9958-812b45371a59.png">

**Filter with multiple tags selected:**
<img width="1171" alt="screen shot 2016-03-18 at 12 47 19 pm" src="https://cloud.githubusercontent.com/assets/3844502/13885910/fab3a70c-ed0a-11e5-9f30-513593cc0eca.png">

**Support for searching for tags:**
<img width="1194" alt="screen shot 2016-03-18 at 1 16 57 pm" src="https://cloud.githubusercontent.com/assets/3844502/13886087/bad5f918-ed0b-11e5-89d6-464013d001f3.png">

**With all three filters being used:**
<img width="1186" alt="screen shot 2016-03-18 at 1 19 20 pm" src="https://cloud.githubusercontent.com/assets/3844502/13886180/14bb4424-ed0c-11e5-94bd-b4297380574d.png">



--

JIRA: https://issues.jboss.org/browse/APIMAN-1025

cc @EricWittmann - Apologies for the delay. I know you didn't want a dropdown but it is included by default, so I wanted to show you how it looks with it. If you'd still prefer not to have it let me know and I'll see if I can hide it with CSS! :)

EricWittmann commented:
I actually *did* want the drop-down, but only if it wasn't a lot of work to implement.  So I'm perfectly happy with that.

Feedback/questions:

1) Sorry to be pedantic - but from the screen shot(s) it looks like the tag filter box is one (1) pixel taller than the other filters (same with the Clear button)
2) Does the impl support clicking on a tag on one of the cards to filter by that tag?

kahboom commented:
@EricWittmann - Great! Glad to hear that; I prefer it with the drop down as well. I fixed the first issue on my side, but then I noticed a difference in font size and padding between the two text boxes is driving me insane so I'm trying to fix it.

As for your second question-- yep! It supports clicking a tag on one of the cards to filter by that tag. The only thing I don't know, and will test now, is whether it will *remove* that tag from the filter if you click it twice. Not sure if that's even something we'd want or is worth implementing if it doesn't.

kahboom commented:
@EricWittmann - See screenshot of new changes:

<img width="907" alt="screen shot 2016-03-18 at 6 27 55 pm" src="https://cloud.githubusercontent.com/assets/3844502/13893759/8c85f302-ed37-11e5-80fc-4b6a7a037bdd.png">


- Adjusted height of input field.
- Removed a box shadow from inside the input field to match the other input field.
- Fixed a CSS issue where the input field would get a blue border on hover or focus; changed it to match the other input field.
- CSS, match font size and padding of other input field.

And just to confirm, clicking on the tag a second time from the card does *not* remove it from the filter, which is the standard and makes sense. Let me know if you'd prefer otherwise though.
